beacon-chain/core/epoch/precompute/attestation.go

```diff
@@ -182,5 +183,28 @@ func UpdateBalance(vp []*Validator, bBal *Balance) *Balance {
 			}
 		}
 	}
+
+	if params.BeaconConfig().EffectiveBalanceIncrement > bBal.ActiveCurrentEpoch {
```

Comment:
can we have a test case specifically for all these conditions ?

Reply:
Clicked the merge too soon @nisdas forgot. there was no branch protection again ugh, we. can perhaps do in a separate PR

Reply:
No prob, we can maybe add it in a separate PR.

Reply:
All good guys! I clicked pushed an hour ago and just realized now the pr has been merged.
Here's the follow up: https://github.com/prysmaticlabs/prysm/pull/6007